### PR TITLE
[ADD] product_firmware_version

### PR DESCRIPTION
**Pull request by bodedra**
#320

### CODE REVIEW

**Review by max3903 — CHANGES_REQUESTED** [review]
@bodedra Please use the OCA logo from https://github.com/OCA/maintainer-tools/blob/master/template/module/static/description/icon.png

**Review by max3903 — APPROVED** [review]

File: product_firmware_version/views/product_view.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    </record>
+    <!-- Product base search view -->
+    <record id="product_template_search_fm_version_view" model="ir.ui.view">
+        <field name="name">product.template.search.view</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_search_view"/>
+        <field name="arch" type="xml">
+            <data>
+                <field name="name" position="attributes">
+                    <attribute name="filter_domain">['|','|','|',('default_code','ilike',self),('name','ilike',self),('barcode','ilike',self), ('firmware_version','ilike',self)]</attribute>
```
Comment (guewen) [inline product_firmware_version/views/product_view.xml]: As soon as another addon will have the same idea about adding a field in this domain, only one of them will work. Wouldn't it be safer to add a new search filter? Or maybe customize the name_search?

The firmware version doesn't seem so important to me to be in the "name" search filter, am I wrong?

Comment (bodedra) [reply product_firmware_version/views/product_view.xml]: Agree.

We have two options for these.

1. Adding new firmware version filter. (User need to select particular these filter option)

2. Override `name_search`. (only append domain filter)

What your suggestion ?

**Review by guewen — APPROVED** [review]
Thanks!

**Review by yvaucher — APPROVED** [review]

**Pull request merged**